backend/src/main/java/com/jobwait/stat/StatRefreshOfficer.java

```diff
@@ -0,0 +1,33 @@
+package com.jobwait.stat;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.jobwait.domain.Stat;
+
+public class StatRefreshOfficer {
+    // constants
+    private static Duration statLifetime = Duration.ofMinutes(60);
+
+    // stat id to timestamp
+    private static Map<String, Instant> generateIdStampMap() {
+        return Stats.known.stream().map(s -> s.id)
+                .collect(Collectors.toMap(Function.identity(), _ -> Instant.now(), (_, _) -> {
+                    throw new RuntimeException("impossible error");
+                }));
+    }
+
+    private Map<String, Instant> statIdToLastRefreshStamp = generateIdStampMap();
+
+    public void refreshStatIfNeeded(String statId) {
```

Comment:
the `IfNeeded` here is a bit ambiguous, what about `refreshOldStat` ? idk just some way to indicate that its only refreshing old stat

Reply:
I wanted to avoid that because the function is not always refreshing, it's only refreshing if it's needed. Is the naming of the function an implementation detail? Maybe just `keepStatsFresh` is abstract enough? I'll change it to that, let me know if you prefer something else